File: multi_car_diary.go
// multi_car_diary.go — Go версия

package main

import (
	"bufio"
	"encoding/json"
	"fmt"
	"os"
	"strconv"
	"strings"
)

type Repair struct {
	Date        string  `json:"date"`
	Mileage     int     `json:"mileage"`
	Description string  `json:"description"`
	Cost        float64 `json:"cost"`
}

type OilChange struct {
	Date        string  `json:"date"`
	Mileage     int     `json:"mileage"`
	OilType     string  `json:"oil_type"`
	Filter      string  `json:"filter"`
	Cost        float64 `json:"cost"`
	Interval    int     `json:"interval"`
	NextMileage int     `json:"next_mileage"`
}

type Reminder struct {
	Title       string `json:"title"`
	Date        string `json:"date"`
	Mileage     int    `json:"mileage"`
	Description string `json:"description"`
	Priority    string `json:"priority"`
	Done        bool   `json:"done"`
}

type Car struct {
	ID          int          `json:"id"`
	Brand       string       `json:"brand"`
	Model       string       `json:"model"`
	Year        int          `json:"year"`
	Vin         string       `json:"vin"`
	Mileage     int          `json:"mileage"`
	Repairs     []Repair     `json:"repairs"`
	OilChanges  []OilChange  `json:"oil_changes"`
	Reminders   []Reminder   `json:"reminders"`
}

type Manager struct {
	Cars []Car `json:"cars"`
	file string
}

func NewManager(file string) *Manager {
	m := &Manager{file: file}
	m.load()
	return m
}

func (m *Manager) load() {
	data, err := os.ReadFile(m.file)
	if err != nil {
		m.Cars = []Car{}
		return
	}
	json.Unmarshal(data, &m.Cars)
}

func (m *Manager) save() {
	data, _ := json.MarshalIndent(m.Cars, "", "  ")
	os.WriteFile(m.file, data, 0644)
}

func (m *Manager) addCar(brand, model string, year int, vin string, mileage int) int {
	id := len(m.Cars) + 1
	m.Cars = append(m.Cars, Car{
		ID:          id,
		Brand:       brand,
		Model:       model,
		Year:        year,
		Vin:         vin,
		Mileage:     mileage,
		Repairs:     []Repair{},
		OilChanges:  []OilChange{},
		Reminders:   []Reminder{},
	})
	m.save()
	return id
}

func (m *Manager) deleteCar(id int) bool {
	for i, c := range m.Cars {
		if c.ID == id {
			m.Cars = append(m.Cars[:i], m.Cars[i+1:]...)
			m.save()
			return true
		}
	}
	return false
}

func (m *Manager) getCar(id int) *Car {
	for i := range m.Cars {
		if m.Cars[i].ID == id {
			return &m.Cars[i]
		}
	}
	return nil
}

func (m *Manager) listCars() {
	if len(m.Cars) == 0 {
		fmt.Println("\u001B[33mНет автомобилей.\u001B[0m")
		return
	}
	fmt.Printf("\u001B[36m%-4s %-15s %-15s %-6s %-10s\u001B[0m\n", "ID", "Марка", "Модель", "Год", "Пробег")
	fmt.Println(strings.Repeat("-", 55))
	for _, c := range m.Cars {
		fmt.Printf("%-4d %-15s %-15s %-6d %-10d\n", c.ID, c.Brand, c.Model, c.Year, c.Mileage)
	}
}

func (m *Manager) carMenu(car *Car) {
	reader := bufio.NewReader(os.Stdin)
	for {
		fmt.Printf("\n\u001B[36m🚗 %s %s %d (Пробег: %d км) — меню автомобиля\u001B[0m\n", car.Brand, car.Model, car.Year, car.Mileage)
		fmt.Println("1. Добавить ремонт")
		fmt.Println("2. Добавить замену масла")
		fmt.Println("3. Добавить напоминание")
		fmt.Println("4. Показать историю")
		fmt.Println("5. Показать статистику")
		fmt.Println("6. Назад")
		fmt.Print("Выберите действие: ")
		choice, _ := reader.ReadString('\n')
		choice = strings.TrimSpace(choice)
		switch choice {
		case "1":
			m.addRepair(car, reader)
		case "2":
			m.addOilChange(car, reader)
		case "3":
			m.addReminder(car, reader)
		case "4":
			m.showHistory(car)
		case "5":
			m.showStats(car)
		case "6":
			return
		default:
			fmt.Println("\u001B[31mНеверный выбор.\u001B[0m")
		}
	}
}

func (m *Manager) addRepair(car *Car, reader *bufio.Reader) {
	fmt.Print("Дата (ГГГГ-ММ-ДД): ")
	date, _ := reader.ReadString('\n')
	date = strings.TrimSpace(date)
	fmt.Print("Пробег (км): ")
	mileageStr, _ := reader.ReadString('\n')
	mileage, _ := strconv.Atoi(strings.TrimSpace(mileageStr))
	fmt.Print("Описание: ")
	desc, _ := reader.ReadString('\n')
	desc = strings.TrimSpace(desc)
	fmt.Print("Стоимость (руб): ")
	costStr, _ := reader.ReadString('\n')
	cost, _ := strconv.ParseFloat(strings.TrimSpace(costStr), 64)

	car.Repairs = append(car.Repairs, Repair{Date: date, Mileage: mileage, Description: desc, Cost: cost})
	if mileage > car.Mileage {
		car.Mileage = mileage
	}
	m.save()
	fmt.Println("\u001B[32m✅ Ремонт добавлен.\u001B[0m")
}

func (m *Manager) addOilChange(car *Car, reader *bufio.Reader) {
	fmt.Print("Дата (ГГГГ-ММ-ДД): ")
	date, _ := reader.ReadString('\n')
	date = strings.TrimSpace(date)
	fmt.Print("Пробег (км): ")
	mileageStr, _ := reader.ReadString('\n')
	mileage, _ := strconv.Atoi(strings.TrimSpace(mileageStr))
	fmt.Print("Тип масла: ")
	oil, _ := reader.ReadString('\n')
	oil = strings.TrimSpace(oil)
	fmt.Print("Фильтр (артикул): ")
	filter, _ := reader.ReadString('\n')
	filter = strings.TrimSpace(filter)
	fmt.Print("Стоимость (руб): ")
	costStr, _ := reader.ReadString('\n')
	cost, _ := strconv.ParseFloat(strings.TrimSpace(costStr), 64)
	fmt.Print("Интервал (км, по умолч. 10000): ")
	intervalStr, _ := reader.ReadString('\n')
	interval := 10000
	if intervalStr = strings.TrimSpace(intervalStr); intervalStr != "" {
		interval, _ = strconv.Atoi(intervalStr)
	}

	nextMileage := mileage + interval
	car.OilChanges = append(car.OilChanges, OilChange{
		Date: date, Mileage: mileage, OilType: oil, Filter: filter,
		Cost: cost, Interval: interval, NextMileage: nextMileage,
	})
	if mileage > car.Mileage {
		car.Mileage = mileage
	}
	m.save()
	fmt.Println("\u001B[32m✅ Замена масла добавлена.\u001B[0m")
}

func (m *Manager) addReminder(car *Car, reader *bufio.Reader) {
	fmt.Print("Название: ")
	title, _ := reader.ReadString('\n')
	title = strings.TrimSpace(title)
	fmt.Print("Дата (ГГГГ-ММ-ДД): ")
	date, _ := reader.ReadString('\n')
	date = strings.TrimSpace(date)
	fmt.Print("Пробег (км): ")
	mileageStr, _ := reader.ReadString('\n')
	mileage, _ := strconv.Atoi(strings.TrimSpace(mileageStr))
	fmt.Print("Описание: ")
	desc, _ := reader.ReadString('\n')
	desc = strings.TrimSpace(desc)
	fmt.Print("Приоритет (низкий/средний/высокий): ")
	priority, _ := reader.ReadString('\n')
	priority = strings.TrimSpace(strings.ToLower(priority))
	if priority != "низкий" && priority != "средний" && priority != "высокий" {
		priority = "средний"
	}
	car.Reminders = append(car.Reminders, Reminder{
		Title: title, Date: date, Mileage: mileage, Description: desc, Priority: priority, Done: false,
	})
	m.save()
	fmt.Println("\u001B[32m✅ Напоминание добавлено.\u001B[0m")
}

func (m *Manager) showHistory(car *Car) {
	fmt.Printf("\n\u001B[36m📋 История для %s %s %d\u001B[0m\n", car.Brand, car.Model, car.Year)
	if len(car.Repairs) > 0 {
		fmt.Println("\u001B[33mРемонты:\u001B[0m")
		for _, r := range car.Repairs {
			fmt.Printf("  %s | %d км | %s | %.2f руб.\n", r.Date, r.Mileage, r.Description, r.Cost)
		}
	}
	if len(car.OilChanges) > 0 {
		fmt.Println("\u001B[33mЗамены масла:\u001B[0m")
		for _, o := range car.OilChanges {
			fmt.Printf("  %s | %d км | %s | %s | %.2f руб. | след. %d км\n", o.Date, o.Mileage, o.OilType, o.Filter, o.Cost, o.NextMileage)
		}
	}
	if len(car.Reminders) > 0 {
		fmt.Println("\u001B[33mНапоминания:\u001B[0m")
		for _, r := range car.Reminders {
			status := "⏳"
			if r.Done {
				status = "✅"
			}
			fmt.Printf("  %s %s | %s | %s | %s\n", status, r.Title, r.Date, r.Priority, r.Description)
		}
	}
}

func (m *Manager) showStats(car *Car) {
	totalRepair := 0.0
	for _, r := range car.Repairs {
		totalRepair += r.Cost
	}
	totalOil := 0.0
	for _, o := range car.OilChanges {
		totalOil += o.Cost
	}
	total := totalRepair + totalOil
	fmt.Printf("\n\u001B[36m📊 Статистика для %s %s %d\u001B[0m\n", car.Brand, car.Model, car.Year)
	fmt.Printf("  Всего ремонтов: %d\n", len(car.Repairs))
	fmt.Printf("  Всего замен масла: %d\n", len(car.OilChanges))
	fmt.Printf("  Всего напоминаний: %d\n", len(car.Reminders))
	fmt.Printf("  Общая стоимость: %.2f руб.\n", total)
	if len(car.Repairs) > 0 {
		fmt.Printf("  Средняя стоимость ремонта: %.2f руб.\n", totalRepair/float64(len(car.Repairs)))
	}
}

func main() {
	manager := NewManager("cars.json")
	reader := bufio.NewReader(os.Stdin)
	for {
		fmt.Println("\n\u001B[36m🚗 Многомарочный дневник автомобиля (Go)\u001B[0m")
		fmt.Println("1. Выбрать/создать автомобиль")
		fmt.Println("2. Удалить автомобиль")
		fmt.Println("3. Показать все автомобили")
		fmt.Println("4. Выход")
		fmt.Print("Выберите действие: ")
		choice, _ := reader.ReadString('\n')
		choice = strings.TrimSpace(choice)
		switch choice {
		case "1":
			manager.listCars()
			if len(manager.Cars) > 0 {
				fmt.Print("Введите номер автомобиля (или 0 для создания нового): ")
				idStr, _ := reader.ReadString('\n')
				idStr = strings.TrimSpace(idStr)
				if idStr == "0" {
					fmt.Print("Марка: ")
					brand, _ := reader.ReadString('\n')
					brand = strings.TrimSpace(brand)
					fmt.Print("Модель: ")
					model, _ := reader.ReadString('\n')
					model = strings.TrimSpace(model)
					fmt.Print("Год: ")
					yearStr, _ := reader.ReadString('\n')
					year, _ := strconv.Atoi(strings.TrimSpace(yearStr))
					fmt.Print("VIN: ")
					vin, _ := reader.ReadString('\n')
					vin = strings.TrimSpace(vin)
					fmt.Print("Текущий пробег (км): ")
					mileageStr, _ := reader.ReadString('\n')
					mileage, _ := strconv.Atoi(strings.TrimSpace(mileageStr))
					id := manager.addCar(brand, model, year, vin, mileage)
					fmt.Printf("\u001B[32m✅ Автомобиль добавлен (ID: %d)\u001B[0m\n", id)
					car := manager.getCar(id)
					if car != nil {
						manager.carMenu(car)
					}
				} else if idStr != "" {
					id, _ := strconv.Atoi(idStr)
					car := manager.getCar(id)
					if car != nil {
						manager.carMenu(car)
					} else {
						fmt.Println("\u001B[31m❌ Автомобиль не найден.\u001B[0m")
					}
				}
			} else {
				fmt.Print("Марка: ")
				brand, _ := reader.ReadString('\n')
				brand = strings.TrimSpace(brand)
				fmt.Print("Модель: ")
				model, _ := reader.ReadString('\n')
				model = strings.TrimSpace(model)
				fmt.Print("Год: ")
				yearStr, _ := reader.ReadString('\n')
				year, _ := strconv.Atoi(strings.TrimSpace(yearStr))
				fmt.Print("VIN: ")
				vin, _ := reader.ReadString('\n')
				vin = strings.TrimSpace(vin)
				fmt.Print("Текущий пробег (км): ")
				mileageStr, _ := reader.ReadString('\n')
				mileage, _ := strconv.Atoi(strings.TrimSpace(mileageStr))
				id := manager.addCar(brand, model, year, vin, mileage)
				fmt.Printf("\u001B[32m✅ Автомобиль добавлен (ID: %d)\u001B[0m\n", id)
				car := manager.getCar(id)
				if car != nil {
					manager.carMenu(car)
				}
			}
		case "2":
			manager.listCars()
			fmt.Print("Введите ID автомобиля для удаления: ")
			idStr, _ := reader.ReadString('\n')
			id, _ := strconv.Atoi(strings.TrimSpace(idStr))
			if manager.deleteCar(id) {
				fmt.Println("\u001B[32m✅ Автомобиль удалён.\u001B[0m")
			} else {
				fmt.Println("\u001B[31m❌ Автомобиль не найден.\u001B[0m")
			}
		case "3":
			manager.listCars()
		case "4":
			fmt.Println("До свидания!")
			return
		default:
			fmt.Println("\u001B[31mНеверный выбор.\u001B[0m")
		}
	}
}

File: multi_car_diary.rs
// multi_car_diary.rs — Rust версия

use serde::{Deserialize, Serialize};
use std::fs;
use std::io::{self, Write};

#[derive(Serialize, Deserialize, Clone)]
struct Repair {
    date: String,
    mileage: u32,
    description: String,
    cost: f64,
}

#[derive(Serialize, Deserialize, Clone)]
struct OilChange {
    date: String,
    mileage: u32,
    oil_type: String,
    filter: String,
    cost: f64,
    interval: u32,
    next_mileage: u32,
}

#[derive(Serialize, Deserialize, Clone)]
struct Reminder {
    title: String,
    date: String,
    mileage: u32,
    description: String,
    priority: String,
    done: bool,
}

#[derive(Serialize, Deserialize, Clone)]
struct Car {
    id: usize,
    brand: String,
    model: String,
    year: u32,
    vin: String,
    mileage: u32,
    repairs: Vec<Repair>,
    oil_changes: Vec<OilChange>,
    reminders: Vec<Reminder>,
}

struct Manager {
    cars: Vec<Car>,
    file: String,
}

impl Manager {
    fn new(file: &str) -> Self {
        let mut m = Manager { cars: Vec::new(), file: file.to_string() };
        m.load();
        m
    }

    fn load(&mut self) {
        if let Ok(data) = fs::read_to_string(&self.file) {
            if let Ok(cars) = serde_json::from_str(&data) {
                self.cars = cars;
                return;
            }
        }
        self.cars = Vec::new();
    }

    fn save(&self) {
        let data = serde_json::to_string_pretty(&self.cars).unwrap();
        fs::write(&self.file, data).unwrap();
    }

    fn add_car(&mut self, brand: String, model: String, year: u32, vin: String, mileage: u32) -> usize {
        let id = self.cars.len() + 1;
        self.cars.push(Car {
            id,
            brand,
            model,
            year,
            vin,
            mileage,
            repairs: Vec::new(),
            oil_changes: Vec::new(),
            reminders: Vec::new(),
        });
        self.save();
        id
    }

    fn delete_car(&mut self, id: usize) -> bool {
        let pos = self.cars.iter().position(|c| c.id == id);
        if let Some(idx) = pos {
            self.cars.remove(idx);
            self.save();
            true
        } else {
            false
        }
    }

    fn get_car(&self, id: usize) -> Option<&Car> {
        self.cars.iter().find(|c| c.id == id)
    }

    fn get_car_mut(&mut self, id: usize) -> Option<&mut Car> {
        self.cars.iter_mut().find(|c| c.id == id)
    }

    fn list_cars(&self) {
        if self.cars.is_empty() {
            println!("\x1b[33mНет автомобилей.\x1b[0m");
            return;
        }
        println!("\x1b[36m{:<4} {:<15} {:<15} {:<6} {:<10}\x1b[0m", "ID", "Марка", "Модель", "Год", "Пробег");
        println!("{}", "-".repeat(55));
        for c in &self.cars {
            println!("{:<4} {:<15} {:<15} {:<6} {:<10}", c.id, c.brand, c.model, c.year, c.mileage);
        }
    }
}

fn main() {
    let mut manager = Manager::new("cars.json");
    loop {
        println!("\n\x1b[36m🚗 Многомарочный дневник автомобиля (Rust)\x1b[0m");
        println!("1. Выбрать/создать автомобиль");
        println!("2. Удалить автомобиль");
        println!("3. Показать все автомобили");
        println!("4. Выход");
        print!("Выберите действие: ");
        io::stdout().flush().unwrap();
        let mut choice = String::new();
        io::stdin().read_line(&mut choice).unwrap();
        match choice.trim() {
            "1" => {
                manager.list_cars();
                if !manager.cars.is_empty() {
                    print!("Введите номер автомобиля (или 0 для создания нового): ");
                    io::stdout().flush().unwrap();
                    let mut input = String::new();
                    io::stdin().read_line(&mut input).unwrap();
                    let input = input.trim();
                    if input == "0" {
                        create_new_car(&mut manager);
                    } else if let Ok(id) = input.parse::<usize>() {
                        if let Some(car) = manager.get_car(id) {
                            car_menu(&mut manager, car.id);
                        } else {
                            println!("\x1b[31m❌ Автомобиль не найден.\x1b[0m");
                        }
                    }
                } else {
                    create_new_car(&mut manager);
                }
            }
            "2" => {
                manager.list_cars();
                print!("Введите ID автомобиля для удаления: ");
                io::stdout().flush().unwrap();
                let mut id_str = String::new();
                io::stdin().read_line(&mut id_str).unwrap();
                if let Ok(id) = id_str.trim().parse::<usize>() {
                    if manager.delete_car(id) {
                        println!("\x1b[32m✅ Автомобиль удалён.\x1b[0m");
                    } else {
                        println!("\x1b[31m❌ Автомобиль не найден.\x1b[0m");
                    }
                }
            }
            "3" => manager.list_cars(),
            "4" => {
                println!("До свидания!");
                break;
            }
            _ => println!("\x1b[31mНеверный выбор.\x1b[0m"),
        }
    }
}

fn create_new_car(manager: &mut Manager) {
    print!("Марка: ");
    io::stdout().flush().unwrap();
    let mut brand = String::new();
    io::stdin().read_line(&mut brand).unwrap();
    let brand = brand.trim().to_string();
    print!("Модель: ");
    io::stdout().flush().unwrap();
    let mut model = String::new();
    io::stdin().read_line(&mut model).unwrap();
    let model = model.trim().to_string();
    print!("Год: ");
    io::stdout().flush().unwrap();
    let mut year_str = String::new();
    io::stdin().read_line(&mut year_str).unwrap();
    let year: u32 = year_str.trim().parse().unwrap();
    print!("VIN: ");
    io::stdout().flush().unwrap();
    let mut vin = String::new();
    io::stdin().read_line(&mut vin).unwrap();
    let vin = vin.trim().to_string();
    print!("Текущий пробег (км): ");
    io::stdout().flush().unwrap();
    let mut mileage_str = String::new();
    io::stdin().read_line(&mut mileage_str).unwrap();
    let mileage: u32 = mileage_str.trim().parse().unwrap();
    let id = manager.add_car(brand, model, year, vin, mileage);
    println!("\x1b[32m✅ Автомобиль добавлен (ID: {})\x1b[0m", id);
    car_menu(manager, id);
}

fn car_menu(manager: &mut Manager, car_id: usize) {
    loop {
        let car = manager.get_car(car_id).unwrap();
        println!("\n\x1b[36m🚗 {} {} {} (Пробег: {} км) — меню автомобиля\x1b[0m", car.brand, car.model, car.year, car.mileage);
        println!("1. Добавить ремонт");
        println!("2. Добавить замену масла");
        println!("3. Добавить напоминание");
        println!("4. Показать историю");
        println!("5. Показать статистику");
        println!("6. Назад");
        print!("Выберите действие: ");
        io::stdout().flush().unwrap();
        let mut choice = String::new();
        io::stdin().read_line(&mut choice).unwrap();
        match choice.trim() {
            "1" => add_repair(manager, car_id),
            "2" => add_oil_change(manager, car_id),
            "3" => add_reminder(manager, car_id),
            "4" => show_history(manager, car_id),
            "5" => show_stats(manager, car_id),
            "6" => break,
            _ => println!("\x1b[31mНеверный выбор.\x1b[0m"),
        }
    }
}

fn add_repair(manager: &mut Manager, car_id: usize) {
    print!("Дата (ГГГГ-ММ-ДД): ");
    io::stdout().flush().unwrap();
    let mut date = String::new();
    io::stdin().read_line(&mut date).unwrap();
    let date = date.trim().to_string();
    print!("Пробег (км): ");
    io::stdout().flush().unwrap();
    let mut mileage_str = String::new();
    io::stdin().read_line(&mut mileage_str).unwrap();
    let mileage: u32 = mileage_str.trim().parse().unwrap();
    print!("Описание: ");
    io::stdout().flush().unwrap();
    let mut desc = String::new();
    io::stdin().read_line(&mut desc).unwrap();
    let desc = desc.trim().to_string();
    print!("Стоимость (руб): ");
    io::stdout().flush().unwrap();
    let mut cost_str = String::new();
    io::stdin().read_line(&mut cost_str).unwrap();
    let cost: f64 = cost_str.trim().parse().unwrap();
    let car = manager.get_car_mut(car_id).unwrap();
    car.repairs.push(Repair { date, mileage, description: desc, cost });
    if mileage > car.mileage { car.mileage = mileage; }
    manager.save();
    println!("\x1b[32m✅ Ремонт добавлен.\x1b[0m");
}

fn add_oil_change(manager: &mut Manager, car_id: usize) {
    print!("Дата (ГГГГ-ММ-ДД): ");
    io::stdout().flush().unwrap();
    let mut date = String::new();
    io::stdin().read_line(&mut date).unwrap();
    let date = date.trim().to_string();
    print!("Пробег (км): ");
    io::stdout().flush().unwrap();
    let mut mileage_str = String::new();
    io::stdin().read_line(&mut mileage_str).unwrap();
    let mileage: u32 = mileage_str.trim().parse().unwrap();
    print!("Тип масла: ");
    io::stdout().flush().unwrap();
    let mut oil = String::new();
    io::stdin().read_line(&mut oil).unwrap();
    let oil = oil.trim().to_string();
    print!("Фильтр (артикул): ");
    io::stdout().flush().unwrap();
    let mut filter = String::new();
    io::stdin().read_line(&mut filter).unwrap();
    let filter = filter.trim().to_string();
    print!("Стоимость (руб): ");
    io::stdout().flush().unwrap();
    let mut cost_str = String::new();
    io::stdin().read_line(&mut cost_str).unwrap();
    let cost: f64 = cost_str.trim().parse().unwrap();
    print!("Интервал (км, по умолч. 10000): ");
    io::stdout().flush().unwrap();
    let mut interval_str = String::new();
    io::stdin().read_line(&mut interval_str).unwrap();
    let interval = if interval_str.trim().is_empty() { 10000 } else { interval_str.trim().parse().unwrap() };
    let next_mileage = mileage + interval;
    let car = manager.get_car_mut(car_id).unwrap();
    car.oil_changes.push(OilChange { date, mileage, oil_type: oil, filter, cost, interval, next_mileage });
    if mileage > car.mileage { car.mileage = mileage; }
    manager.save();
    println!("\x1b[32m✅ Замена масла добавлена.\x1b[0m");
}

fn add_reminder(manager: &mut Manager, car_id: usize) {
    print!("Название: ");
    io::stdout().flush().unwrap();
    let mut title = String::new();
    io::stdin().read_line(&mut title).unwrap();
    let title = title.trim().to_string();
    print!("Дата (ГГГГ-ММ-ДД): ");
    io::stdout().flush().unwrap();
    let mut date = String::new();
    io::stdin().read_line(&mut date).unwrap();
    let date = date.trim().to_string();
    print!("Пробег (км): ");
    io::stdout().flush().unwrap();
    let mut mileage_str = String::new();
    io::stdin().read_line(&mut mileage_str).unwrap();
    let mileage: u32 = mileage_str.trim().parse().unwrap();
    print!("Описание: ");
    io::stdout().flush().unwrap();
    let mut desc = String::new();
    io::stdin().read_line(&mut desc).unwrap();
    let desc = desc.trim().to_string();
    print!("Приоритет (низкий/средний/высокий): ");
    io::stdout().flush().unwrap();
    let mut priority = String::new();
    io::stdin().read_line(&mut priority).unwrap();
    let priority = priority.trim().to_lowercase();
    let priority = if priority == "низкий" || priority == "средний" || priority == "высокий" { priority } else { "средний".to_string() };
    let car = manager.get_car_mut(car_id).unwrap();
    car.reminders.push(Reminder { title, date, mileage, description: desc, priority, done: false });
    manager.save();
    println!("\x1b[32m✅ Напоминание добавлено.\x1b[0m");
}

fn show_history(manager: &Manager, car_id: usize) {
    let car = manager.get_car(car_id).unwrap();
    println!("\n\x1b[36m📋 История для {} {} {}\x1b[0m", car.brand, car.model, car.year);
    if !car.repairs.is_empty() {
        println!("\x1b[33mРемонты:\x1b[0m");
        for r in &car.repairs {
            println!("  {} | {} км | {} | {} руб.", r.date, r.mileage, r.description, r.cost);
        }
    }
    if !car.oil_changes.is_empty() {
        println!("\x1b[33mЗамены масла:\x1b[0m");
        for o in &car.oil_changes {
            println!("  {} | {} км | {} | {} | {} руб. | след. {} км", o.date, o.mileage, o.oil_type, o.filter, o.cost, o.next_mileage);
        }
    }
    if !car.reminders.is_empty() {
        println!("\x1b[33mНапоминания:\x1b[0m");
        for r in &car.reminders {
            let status = if r.done { "✅" } else { "⏳" };
            println!("  {} {} | {} | {} | {}", status, r.title, r.date, r.priority, r.description);
        }
    }
}

fn show_stats(manager: &Manager, car_id: usize) {
    let car = manager.get_car(car_id).unwrap();
    let total_repair: f64 = car.repairs.iter().map(|r| r.cost).sum();
    let total_oil: f64 = car.oil_changes.iter().map(|o| o.cost).sum();
    let total = total_repair + total_oil;
    println!("\n\x1b[36m📊 Статистика для {} {} {}\x1b[0m", car.brand, car.model, car.year);
    println!("  Всего ремонтов: {}", car.repairs.len());
    println!("  Всего замен масла: {}", car.oil_changes.len());
    println!("  Всего напоминаний: {}", car.reminders.len());
    println!("  Общая стоимость: {:.2} руб.", total);
    if !car.repairs.is_empty() {
        println!("  Средняя стоимость ремонта: {:.2} руб.", total_repair / car.repairs.len() as f64);
    }
}
